lib/reek/smells/smell_detector.rb

```diff
@@ -38,10 +38,19 @@ def contexts
       end
 
       def run_for(context)
-        return [] unless enabled_for?(context)
         return [] if exception?(context)
 
-        sniff(context)
+        smells = sniff(context)
+
+        if enabled_for?(context)
```

Comment:
I'm a big fan of early returns instead of if/else's but I think in this case @mvz first suggestion:

```Ruby
if enabled_for? context
  sniff(context)
else
  check_for_unneeded_suppression(context)
end
```

is way more readable.